nilearn/plotting/displays/_figures.py

```diff
@@ -139,7 +140,6 @@ def add_contours(
         labels=None,
         lines=None,
         elevation=0.1,
-        hemi="left",
```

Comment:
Setting `hemi` via `add_contours` method could allow plotting contours for right hemisphere on the mesh for left hemisphere.

I removed it from here, and it is now passed into the `SurfaceFigure` object so that it can be tracked consistently across all methods.

See commit https://github.com/nilearn/nilearn/pull/4991/commits/3647cb5a98b0622e41e124abc739379e76875696 for easier review